### DISCUSSION TITLE
Get Active Design

### DISCUSSION THREAD

**Discussion by hmulhi** [ansys/pyaedt#3821] [Q&A] [answered]
I have a list of Q3D designs and if I use the default `pyaedt.q3d.Q3d(projectname=None, designname=None)`, the active project is correctly identified but a new q3d design is created. In the documentation, it is stated that an attempt is made to get the active design, but it never worked for me.

My question is: Is there a way to reliably get the active design without having to specify it's name? Something equivalent to:
```
oProject = oDesktop.GetActiveProject()
oDesign = oProject.GetActiveDesign()
```

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @hmulhi ,

This is expected, because if you have more than one Q3D design, PyAEDT cannot decide which one you want to connect. So PyAEDT created a new design. If you have this kind of situation, I encourage you to use the first Desktop, then you will know the available projects and designs in the AEDT session.
Then for example you can use: design_list method.

app = pyaedt.Desktop()
app.load_project("project_path")
app.design_list()

Then you can use pyaedt.Hfss(designname=...) 

or even

app[("projectname", "designname")]

**Reply by hmulhi** [reply]
Hi @Samuelopez-ansys ,

thanks for the quick response!

Well I wanted PyAEDT to choose the active design of the active project, so that I can write generalized scripts that performs tasks for any design I am working on.
I came up with the following work-around. It gets the job done, but it is inefficient:

```python
from pyaedt import Desktop, Q3d
window = Desktop(specified_version="2022.2", new_desktop_session=False, close_on_exit=False)
project = window.odesktop.GetActiveProject()
design = project.GetActiveDesign()
q3dDesignName = design.GetName()
Q3D_design = Q3d(designname=q3dDesignName, specified_version="2022.2")
```

Anyways, I understand from your answer that it is at the moment not possible for PyAEDT to directly fitch the active design. Maybe it would be helpful to reconsider the following line in the documentation:

> designname[str], optional
Name of the design to select. The default is None, in which case an attempt is made to get an active design. If no designs are present, an empty design is created.

thanks again :)
